### PR TITLE
Add test cases for File Block

### PR DESCRIPTION
**Pull request by jd-alexander**
This PR added the test cases for the File Block. 

To view a preview of the markdown document click [here](https://github.com/wordpress-mobile/test-cases/blob/5482b3428993ef856c1339bf25517279b4f5db4e/test-cases/gutenberg/file.md)

#### Questions

* I saw other test cases with preconditions. Does the File block have any of those?
* I included a few images in the test cases. Let me know if you have any thoughts on them. Also, let me know if the image sizes are fine for this purpose.

### CODE REVIEW

**Comment by hypest** [comment]
Oops, the `.DS_Store` file was included in the commit, let's remove it and add a .gitignore to help us keep it out next time 👍

File: test-cases/gutenberg/file.md [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+##### TC002
+
+**Upload completes successfully from the WordPress Media library**
+
+**Steps**
+- Tap on Choose a file on the Media Placeholder. 
+- Tap option "WordPress Media Library" and choose a file. 
+
+**Expected Behavior**
+- Expect to see file's name  (without link)  and an active Download button.
```
Comment (etoledom) [inline test-cases/gutenberg/file.md:32]: What does it mean `without link`?

File: test-cases/gutenberg/file.md [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+**Steps**
+
+- Press the Block settings or Link Settings icon.
+- Press the Link to icon. 
+
+**Expected Behavior**
+
+- Two different options should appear: Media file and Attachment Page
+- Media File: The link should be set to the file itself.
+- Attachment page: The link should be set to the item page.
```
Comment (etoledom) [inline test-cases/gutenberg/file.md:175]: I would add a step to check the preview site that the proper link was set and it is not broken.
Even though there's that known issue about recently uploaded files that will change their attachment url after autosaving on mobile.

File: test-cases/gutenberg/file.md [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+**Expected Behavior**
+
+-   You should see that File block is cleared and the placeholder component is now visible.
+![Upload placeholder](../resources/local-file-upload-placeholder.png)
+
+--------------------------------------------------------------------------------
+##### TC005
+
+**Close the post with an ongoing file upload**
+
```
Comment (etoledom) [inline test-cases/gutenberg/file.md:103]: I would also add the test case of closing the editor and opening it again while a file is being updated.
Expected:
- The block should detect the ongoing upload and show the progress bar.

File: test-cases/gutenberg/file.md [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+- Expect to see file's name and an active Download button once the upload is complete.  
+- Expect the link and replace buttons to appear in the toolbar
+
+![Upload successful](../resources/local-file-upload-successful.png)
+
+--------------------------------------------------------------------------------
+
+##### TC002
+
+**Upload completes successfully from the WordPress Media library**
```
Comment (etoledom) [inline test-cases/gutenberg/file.md:25]: Nitpick: Adding from WordPress Media Library is technically not an Upload. It's just inserted.

**Review by etoledom — COMMENTED** [review]
Great job @jd-alexander - I left a few comments but overall it's a great job! 🎉

File: test-cases/gutenberg/file.md (hunk 2 above)
Comment (jd-alexander) [reply test-cases/gutenberg/file.md:175]: Sounds good!

File: test-cases/gutenberg/file.md (hunk 1 above)
Comment (jd-alexander) [reply test-cases/gutenberg/file.md]: Removed here bd426a5

File: test-cases/gutenberg/file.md (hunk 3 above)
Comment (jd-alexander) [reply test-cases/gutenberg/file.md:103]: Done 30d7541

File: test-cases/gutenberg/file.md (hunk 4 above)
Comment (jd-alexander) [reply test-cases/gutenberg/file.md]: Fixed in bfad517 along with placeholder changes we had discussed.

**Comment by jd-alexander** [comment]
Thanks for the thorough review @etoledom 😄  This is ready for another round 👍

File: test-cases/gutenberg/file.md [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+
+--------------------------------------------------------------------------------
+##### TC012
+
+**Close/Re-open post with an ongoing file upload**
+
+**Steps**
+
+-   Add a File block and insert a file from your device.
+-   While there’s an ongoing upload, save the post as a draft.
```
Comment (etoledom) [inline test-cases/gutenberg/file.md:230]: I think we should note that closing the editor is a necessary step here too.

```suggestion
-   While there’s an ongoing upload, close the editor by saving the post as draft.
```

**Comment by etoledom** [comment]
Thank you for the update! Looking great 🎉 
Just one small extra detail and we are ready to go 🙏

**Comment by jd-alexander** [comment]
Thanks @etoledom It's ready to go!

**Review by etoledom — APPROVED** [review]
:shipit: 🎉

**Pull request merged**